### ISSUE TITLE
Support static calls to `#[Scope]` attribute methods

### ISSUE THREAD

**Issue by alies-dev** [psalm/psalm-plugin-laravel#634]
## Problem

When a model method has the `#[Scope]` attribute, Laravel handles static calls via `Model::__callStatic()` → `query()` → `Builder::__call()` → `callNamedScope()`. So `Post::popular()` works at runtime.

However, Psalm sees `popular()` as a real instance method and reports `InvalidStaticInvocation` before any plugin handler gets a chance to intercept. This is a false positive.

```php
#[Scope]
public function popular(Builder $query): void { ... }

Post::popular(); // ❌ InvalidStaticInvocation — works at runtime
Post::query()->popular(); // ✅ Fixed in #633
```

Legacy scopes (`scopeActive` → `Post::active()`) work because `active()` doesn't exist as a real method — the plugin's existence handler confirms it. `#[Scope]` methods are different because they exist as real instance methods.

## Suggested fix

Implement `BeforeAddIssueInterface` to suppress `InvalidStaticInvocation` for `#[Scope]` methods on Model subclasses. The detection logic already exists in `BuilderScopeHandler::hasScopeMethod()`.

~40 lines in a new handler + registration in `Plugin.php` + test updates.

## Context

Discovered during #633. Currently documented as a known limitation in both `CustomQueryBuilderTest.phpt` and `ModelStaticBuilderMethodsTest.phpt`.

**Comment by alies-dev** [comment]
Status after the v4.13.x scope work, with the breakdown of what is and isn't fixable at the plugin level.

This splits into two cases by the scope method's visibility, because PHP's static-dispatch rule differs:

**Public `#[Scope]` method called statically** (e.g. `Post::popular()` where `popular` is `public`) — the `InvalidStaticInvocation` is a **true positive**, not a bug. A `public` non-static method called statically is a fatal `Error` in PHP; dispatch never reaches `__callStatic`, so the scope is never invoked at runtime. The diagnostic is correct here and should stay.

**Protected `#[Scope]` method called statically** (Laravel's documented convention) — this is the real false positive. The method is inaccessible from the calling scope, so PHP routes the static call through `Model::__callStatic` → `static::query()->scope()`, which is runtime-valid. Psalm still reports `InvalidStaticInvocation` because `MethodAnalyzer::checkStatic` flags any static call to a non-static method without considering accessibility plus the presence of `__callStatic`.

That last part is an upstream Psalm limitation, tracked at **https://github.com/vimeo/psalm/issues/11876**. It is not cleanly fixable at the plugin level: the only available lever (a `BeforeAddIssue` handler) could suppress the false positive but cannot also supply the call's return type and parameters, so it would be a half-fix. Once #11876 is resolved, the plugin can resolve the static-call type to `Builder<Model>` with the scope's parameters, mirroring what it already does for the instance and forwarded forms.

Summary: the public-static case is intended behavior; the protected-static case is blocked on vimeo/psalm#11876. Leaving this open to track the latter.

**Comment by alies-dev** [comment]
Closing as not actionable at the plugin level. Two cases:

**Public `#[Scope]` called statically** (`Post::popular()`, `popular` is public) — `InvalidStaticInvocation` is a **true positive**. A public non-static method called statically is a fatal PHP `Error`; dispatch never reaches `__callStatic`, so the scope never runs. Diagnostic is correct.

**Protected `#[Scope]` called statically** (Laravel's documented convention) — real false positive, but blocked upstream: `MethodAnalyzer::checkStatic` flags any static call to a non-static method without accounting for accessibility + `__callStatic`. Tracked at **vimeo/psalm#11876**. The only plugin lever (`BeforeAddIssue`) could suppress the error but can't also supply the call's return type/params, so it'd be a half-fix.

Reopen once vimeo/psalm#11876 lands — the plugin can then resolve the protected-static call to `Builder<Model>` with the scope's params, mirroring the instance/forwarded forms it already handles.
